From a4ce96735ad8f1e07ded93e39e32e22bd9ac00ba Mon Sep 17 00:00:00 2001
From: Cornelia Huck <cohuck@redhat.com>
Date: Tue, 1 Dec 2020 17:53:41 -0500
Subject: redhat: s390x: add rhel-8.4.0 compat machine

RH-Author: Cornelia Huck <cohuck@redhat.com>
Message-id: <20201201175341.37537-3-cohuck@redhat.com>
Patchwork-id: 100195
O-Subject: [RHEL-AV-8.4.0 qemu-kvm PATCH 2/2] redhat: s390x: add rhel-8.4.0 compat machine
Bugzilla: 1836282
RH-Acked-by: Miroslav Rezanina <mrezanin@redhat.com>
RH-Acked-by: Thomas Huth <thuth@redhat.com>
RH-Acked-by: David Hildenbrand <david@redhat.com>

Note that we did not publish a rhel-8.3.0 machine on s390x, so we
need to add the respective hw_compat entry in the rhel-8.2.0 machine.

Also, the hw_compat entry for 8.1 was missing; however, the contents
there are not relevant for s390x.

Signed-off-by: Cornelia Huck <cohuck@redhat.com>
Signed-off-by: Danilo C. L. de Paula <ddepaula@redhat.com>
---
 hw/s390x/s390-virtio-ccw.c | 17 ++++++++++++++++-
 1 file changed, 16 insertions(+), 1 deletion(-)

diff --git a/hw/s390x/s390-virtio-ccw.c b/hw/s390x/s390-virtio-ccw.c
index b8dde7e4e1..c7b5bcb06b 100644
--- a/hw/s390x/s390-virtio-ccw.c
+++ b/hw/s390x/s390-virtio-ccw.c
@@ -1056,15 +1056,29 @@ static void ccw_machine_2_4_class_options(MachineClass *mc)
 DEFINE_CCW_MACHINE(2_4, "2.4", false);
 #endif
 
+static void ccw_machine_rhel840_instance_options(MachineState *machine)
+{
+}
+
+static void ccw_machine_rhel840_class_options(MachineClass *mc)
+{
+}
+DEFINE_CCW_MACHINE(rhel840, "rhel8.4.0", true);
+
 static void ccw_machine_rhel820_instance_options(MachineState *machine)
 {
+    ccw_machine_rhel840_instance_options(machine);
 }
 
 static void ccw_machine_rhel820_class_options(MachineClass *mc)
 {
+    ccw_machine_rhel840_class_options(mc);
     mc->fixup_ram_size = s390_fixup_ram_size;
+    /* we did not publish a rhel8.3.0 machine */
+    compat_props_add(mc->compat_props, hw_compat_rhel_8_3, hw_compat_rhel_8_3_len);
+    compat_props_add(mc->compat_props, hw_compat_rhel_8_2, hw_compat_rhel_8_2_len);
 }
-DEFINE_CCW_MACHINE(rhel820, "rhel8.2.0", true);
+DEFINE_CCW_MACHINE(rhel820, "rhel8.2.0", false);
 
 static void ccw_machine_rhel760_instance_options(MachineState *machine)
 {
@@ -1086,6 +1100,7 @@ static void ccw_machine_rhel760_class_options(MachineClass *mc)
 {
     ccw_machine_rhel820_class_options(mc);
     /* We never published the s390x version of RHEL-AV 8.0 and 8.1, so add this here */
+    compat_props_add(mc->compat_props, hw_compat_rhel_8_1, hw_compat_rhel_8_1_len);
     compat_props_add(mc->compat_props, hw_compat_rhel_8_0, hw_compat_rhel_8_0_len);
     compat_props_add(mc->compat_props, hw_compat_rhel_7_6, hw_compat_rhel_7_6_len);
 }
-- 
2.18.4

